packages/serialize/src/lib/Serialize.ts

```diff
@@ -90,62 +104,80 @@ export namespace Serialize {
   }
 
   function fromJSONArray(json: JSON[]): unknown[] {
-    return json.reduce<unknown[]>((raw, value) => [...raw, fromJSON(value)], []);
+    const arr: unknown[] = [];
+
+    for (const value of json) arr.push(fromJSON(value));
+    return arr;
   }
 
   function fromJSONMap(json: [PropertyKey, JSON][]): [PropertyKey, unknown][] {
-    return json.reduce<[PropertyKey, unknown][]>((raw, [key, value]) => [...raw, [key, fromJSON(value)]], []);
+    const arr: [PropertyKey, unknown][] = [];
+
+    for (const [key, value] of json) arr.push([key, fromJSON(value)]);
+    return arr;
   }
 
   function fromJSONObject(json: Record<PropertyKey, JSON>): Record<PropertyKey, unknown> {
-    return Object.entries(json).reduce<Record<PropertyKey, unknown>>((raw, [key, value]) => ({ ...raw, [key]: fromJSON(value) }), {});
+    const obj: Record<PropertyKey, unknown> = {};
+
+    for (const key in json) {
+      if ({}.hasOwnProperty.call(json, key)) {
```

Comment:
```suggestion
      if (json.hasOwnProperty(key)) {
```

Reply:
[This](https://stackoverflow.com/a/56024114) for why not